cmake/common.cmake

```diff
@@ -48,6 +48,59 @@ macro(irr_create_executable_project _EXTRA_SOURCES _EXTRA_OPTIONS _EXTRA_INCLUDE
 	endif()
 endmacro()
 
+macro(irr_create_library_project EXECUTABLE_NAME _EXTRA_SOURCES _EXTRA_OPTIONS _EXTRA_INCLUDES _EXTRA_LIBS)
+	project(${EXECUTABLE_NAME})
+
+	add_library(${EXECUTABLE_NAME} ${_EXTRA_SOURCES})
+	# EXTRA_SOURCES is var containing non-common names of sources (if any such sources, then EXTRA_SOURCES must be set before including this cmake code)
+	add_dependencies(${EXECUTABLE_NAME} Irrlicht)
+
+	target_include_directories(${EXECUTABLE_NAME}
+		PUBLIC ../../include
+		PRIVATE ${_EXTRA_INCLUDES}
+	)
+	target_link_libraries(${EXECUTABLE_NAME} Irrlicht ${_EXTRA_LIBS})
+	target_compile_options(${EXECUTABLE_NAME} PUBLIC ${_EXTRA_OPTIONS})
+
+	if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+		add_compile_options(
+			"$<$<CONFIG:DEBUG>:-fstack-protector-all>"
+		)
+
+		set(COMMON_LINKER_OPTIONS "-msse4.2 -mfpmath=sse -fuse-ld=gold")
+		set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${COMMON_LINKER_OPTIONS}")
+		set(CMAKE_EXE_LINKER_FLAGS_DEBUG "${COMMON_LINKER_OPTIONS} -fstack-protector-strong -fsanitize=address")
```

Comment:
those are only enabled by a special IRRBAW option when building the main library, because they inject into process and make it impossible to use Renderdoc

maybe you could hijack the main irrlibrary compile options (they are separate and different for Relaese and Debug targets)

Reply:
Could you tell me the involved line numbers?

Reply:
we actually have these beautiful options https://github.com/buildaworldnet/IrrlichtBAW/blob/master/CMakeLists.txt#L32